feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+                    optimization_method: str, cost_metric: str) -> NoReturn:
+        """_process_df
+        We have implemented two versions of this function. In the version here, the categories parameter
+        actually refers to the labels (rather than categories) to be covered. It generates features for input data
+        using a specified featurization method, and creates a matrix where each row corresponds to a label and each
+        column corresponds to a feature.
+        The version of code for generating labels from categories has been moved to utils.py.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: List[int]
+            List of columns in data that contains categories/labels to be covered.
+
+        selection_size: int
+            Number of feature to select.
+
+        optimization_method: str
+            The function need method to prevent performing featurization in
+             the case that optimization method is random(t or unicost) or greedy(unicost)
+
+        cost_metric: str
+            The function need cost metric to prevent performing featurization in
+             the case that cost metric is None or unicost (for random and greedy)
+
+        Returns
+        -------
+        No return
+       --------
+        """
+        self.matrix = categories.to_numpy()
+        self._num_rows, self._num_cols = self.matrix.shape
+        option_map = {
+            ("random", True, "diverse"): lambda: None,
+            ("random", False, "unicost"): lambda: None,
+            ("random", True, "unicost"): lambda: None,
+            ("greedy", False, "unicost"): lambda: None,
+            ("greedy", True, "unicost"): lambda: None,
+            ("exact", True, "unicost"): lambda: None,
+            ("exact", False, "unicost"): lambda: None
+        }
+
+        options = option_map.get((optimization_method, selection_size is not None, cost_metric))
+        if options:
+            options()
+        else:
+            feature_column = self.featurization_method.fit_transform(input_df)
+            self.features = np.array([eval(l) if isinstance(l, str) else l for l in feature_column.tolist()])
+            check_true(len(self.features) == self._num_cols,
+                       f"Process Data Error: features size ({len(self.features)}) "
+                       f"should match the number of columns ({self._num_cols})")
+
+        check_true(self.matrix.ndim == 2, "Process Data Error: matrix should 2D")
+        if self.selection_size is not None:
+            check_true(self.selection_size <= self._num_cols,
+                       "Process Data Error: selection_size cannot exceed num columns")
+
+    def _select_multi_level_optimization(self) -> List:
+        unicost = np.ones(self._num_cols)
+        if self.selection_size is None:
```

Comment:
cannot use get_selection_size()

Reply:
This comment is not clear to me. Would you please explain more?

Reply:
I meant cannot we re-use get_selection_size() here